news-explorer/src/components/SearchForm/SearchForm.js

```diff
@@ -1,26 +1,56 @@
 import React, { useContext } from 'react';
 import './SearchForm.css';
 import { SmallScreenContext } from '../../contexts/SmallScreenContext';
+import { useFormValidation } from '../../utils/useFormValidation.js';
 
-const SearchForm = () => {
+
+const SearchForm = ({ sendSearchQuery, setInfoPopup, setPreLoader }) => {
   const { isSmallScreen } = useContext(SmallScreenContext);
-  
+  const {
+    values, handleChange,
+  } = useFormValidation();
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    const q = values.newsSearch.trim();
+    if (q === '') { setInfoPopup({ isInfoOpen: true, msg: 'Please enter a keyword', displayLink: false }); return; }
+    setPreLoader({ isLoading: true, hasResults: true });
+    sendSearchQuery(q);
+  };
+
   return (
-    <form className="search-form">
+    <form className="search-form" onSubmit={handleSubmit}>
       <div className="search-form__txt-container">
         <h1 className='search-form__heading'>What's going on in the world?</h1>
         <p className='search-form__description'>Find the latest news on any topic and save them in your personal account.</p>
       </div>
       {!isSmallScreen ? (<>
         <div className="search-form__search-bar">
-          <input className="search-form__input" type="text" placeholder="Enter topic" />
-          <button className='search-form__submit'>Search</button>
+          <input className="search-form__input"
+            type="search"
+            required
+            placeholder="Enter topic"
+            id='search'
+            name='newsSearch'
+            onChange={handleChange}

```

Comment:
If you have `onChange` or ` onInput` in an input then you need to add attribute `value` to this input.
This is how React controls inputs.

To read more about **controlled components**: https://reactjs.org/docs/forms.html#controlled-components